engine/src/input/axis.cpp

```diff
@@ -60,11 +60,36 @@ float InputAxis::value() const
 
 void InputAxis::value(float value)
 {
-    mValue = value;
+    if (std::abs(value) < mDeadzone)
+    {
+        mValue = 0.0F;
+    }
+    else
+    {
+        mValue = value;
+    }
```

Comment:
If you want, you could make it in a ternary operator, in my opinion, it is not complex logic nor unreadable in a single line.
`mValue = std::abs(value) < mDeadzone ? 0.0F : value;`